File: src/lib/types/badge.ts
export interface Badge {
	id: string;
	program_id: string;
	badge_type: string;
	title: string;
	description: string;
	icon: string;
	earned_at: string;
	metadata: string;
}

export interface BadgeProgress {
	badge_type: string;
	title: string;
	description: string;
	current_value: number;
	target_value: number;
	progress_percent: number;
	is_earned: boolean;
}

File: src-tauri/src/db/models/badge.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct Badge {
    pub id: String,
    pub program_id: String,
    pub badge_type: String,
    pub title: String,
    pub description: String,
    pub icon: String,
    pub earned_at: String,
    pub metadata: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct BadgeProgress {
    pub badge_type: String,
    pub title: String,
    pub description: String,
    pub current_value: i32,
    pub target_value: i32,
    pub progress_percent: f64,
    pub is_earned: bool,
}
